src/ai/backend/client/func/quota_scope.py

```diff
@@ -0,0 +1,156 @@
+        domain_name: str,
+        email: str,
+        fields: Sequence[FieldSpec] = _default_user_fields,
+    ):
+        query = textwrap.dedent(
+            """\
+            query($domain_name: String!, $email: String!) {
+                user(domain_name: $domain_name, email: $email) {$fields}
+            }
+        """
+        )
+        query = query.replace("$fields", " ".join(f.field_ref for f in fields))
+        variables = {
+            "domain_name": domain_name,
+            "email": email,
+        }
+        data = await api_session.get().Admin._query(query, variables)
+        return data["user"]
+
+    @api_function
+    @classmethod
+    async def get_project_qsid(
+        cls,
+        domain_name: str,
+        name: str,
+        fields: Sequence[FieldSpec] = _default_project_fields,
+    ):
+        query = textwrap.dedent(
+            """\
+            query($domain_name: String!, $name: String!) {
+                groups_by_name(domain_name: $domain_name, name: $name) {$fields}
+            }
+        """
+        )
+        query = query.replace("$fields", " ".join(f.field_ref for f in fields))
+        variables = {
+            "domain_name": domain_name,
+            "name": name,
+        }
+        data = await api_session.get().Admin._query(query, variables)
+        return data["groups_by_name"][0]
+
+    @api_function
+    @classmethod
+    async def get_quota_scope(
+        cls,
+        host: str,
+        qsid: str,
+        fields: Sequence[FieldSpec] = _default_detail_fields,
+    ):
+        query = textwrap.dedent(
+            """\
+            query($storage_host_name: String!, $quota_scope_id: String!) {
+                quota_scope(storage_host_name: $storage_host_name, quota_scope_id: $quota_scope_id) {
+                    details {$fields}
+                }
+            }
+        """
+        )
+        query = query.replace("$fields", " ".join(f.field_ref for f in fields))
+        variables = {
+            "storage_host_name": host,
+            "quota_scope_id": qsid,
+        }
+        data = await api_session.get().Admin._query(query, variables)
+        return data["quota_scope"]["details"]
+
+    @api_function
+    @classmethod
+    async def set_quota_scope(
+        cls,
+        host: str,
+        qsid: str,
+        hard_limit_bytes: int,
+        fields: Sequence[FieldSpec] = _default_quota_scope_fields,
+    ):
+        query = textwrap.dedent(
+            """\
+            mutation($storage_host_name: String!, $quota_scope_id: String!, $input: QuotaScopeInput!) {
+                set_quota_scope(storage_host_name: $storage_host_name, quota_scope_id: $quota_scope_id, props: $input) {
+                    quota_scope {$fields}
+                }
+            }
+        """
+        )
+        query = query.replace("$fields", " ".join(f.field_ref for f in fields))
+        inputs: dict[str, Any] = {}
+        set_if_set(inputs, "hard_limit_bytes", hard_limit_bytes)
```

Comment:
we don't need to use `set_if_set` here because `hard_limit_bytes` should be `None` or `Undefined` here.

Reply:
> we don't need to use `set_if_set` here because `hard_limit_bytes` should be `None` or `Undefined` here.

I'm not sure I fully understand your comment, but isn't `hard_limit_bytes` not `None` or `Undefined` here? Because `hard_limit_bytes` is required parameter in set CLI command.

Reply:
Sorry for my misspelling. `hard_limit_bytes` should **NOT** be `None` or `Undefined`